src/pal/src/misc/time.cpp

```diff
@@ -270,6 +275,29 @@ GetTickCount64()
         }
         retval = (ts.tv_sec * tccSecondsToMillieSeconds)+(ts.tv_nsec / tccMillieSecondsToNanoSeconds);
     }
+#elif HAVE_MACH_CLOCK_MONOTONIC
+    {
+        kern_return_t machRet;
+        clock_serv_t clock;
+        mach_timespec_t mts;
+        host_t host = mach_host_self();
+
+        if((machRet = host_get_clock_service(host, SYSTEM_CLOCK, &clock)) != KERN_SUCCESS)
```

Comment:
@jkotas I did some crude performance measurements around Environment.TickCount.
With my changes (no caching) - Average time 3.0 microseconds
Without my changes (using gettimeofday) - Average time 0.13 microseconds.
I am not sure if perf cost is within acceptable level of tolerance. Meanwhile I am also trying to get perf measurements with caching the clock service.

It did my testing on release builds and got the average from 10 million iterations.
```c#
[Fact]
        public static void TickCountPerf()
        {            
            int iterations = 10000000;
            long totalTime = 0;
            for (int i = 0; i < iterations; i++)
            {
                long startTime = GetTimeofDay();
                int tick = Environment.TickCount;
                totalTime += GetTimeofDay() - startTime;
            }

            float average = (float)totalTime / iterations;
            Console.WriteLine("totalTime - {0} usecs", totalTime);
            Console.WriteLine("Average - {0} usecs", average);
        }

        public static long GetTimeofDay()
        {
            timeval tv = new timeval();           
            int result = gettimeofday(ref tv, IntPtr.Zero);            
            return (tv.tv_sec * tccSecondsToMicroSeconds) + tv.tv_usec;
        }

        [DllImport("libc")]
        // provides microsecond level resolution
        private static extern int gettimeofday(ref timeval tv, IntPtr tz);
```